tests/data_group_test.py

```diff
@@ -207,24 +209,215 @@ def test_getitem_label_indexing_based_works_when_length_is_not_unique():
+    assert 'a' in result
+    assert sc.identical(result['a'], op(x, other))
+
+
+@pytest.mark.parametrize('op', BINARY_OPERATIONS, ids=BINARY_OPERATION_NAMES)
+@pytest.mark.parametrize('reverse', (False, True))
+def test_arithmetic_data_group_with_data_array(op, reverse):
+    op = reverse_op(op, reverse)
+    x = sc.DataArray(sc.arange('x', 1, 5, unit='m'), coords={'x': sc.arange('x', 5)})
+    dg = sc.DataGroup({'a': x})
+    other = 2 * x
+    result = op(dg, other)
+    assert 'a' in result
+    assert sc.identical(result['a'], op(x, other))
+
+
+@pytest.mark.parametrize('op', BINARY_OPERATIONS, ids=BINARY_OPERATION_NAMES)
+def test_arithmetic_data_group_with_data_array_coord_mismatch(op):
+    x = sc.DataArray(sc.arange('x', 1, 5, unit='m'), coords={'x': sc.arange('x', 5)})
+    dg = sc.DataGroup({'a': x.copy()})
+    other = 2 * x
+    other.coords['x'][0] = -1
+    with pytest.raises(sc.DatasetError):
+        op(dg, other)
+
+
+def test_pow_data_group_with_data_group():
+    x = sc.arange('x', 1, 5)
+    dg1 = sc.DataGroup({'a': x})
+    dg2 = sc.DataGroup({'a': 2 * x, 'b': x})
+    result = dg1**dg2
+    assert 'a' in result
+    assert 'b' not in result
+    assert sc.identical(result['a'], x**(2 * x))
+
+
+@pytest.mark.parametrize('other', (3, sc.arange(
+    'x', 0, 4), sc.DataArray(sc.arange('x', 0, 4), coords={'x': sc.arange('x', 5)})),
+                         ids=('int', 'Variable', 'DataArray'))
+@pytest.mark.parametrize('reverse', (False, True))
+def test_pow_data_group_with_other(other, reverse):
+    op = reverse_op(lambda a, b: a**b, reverse)
+    x = sc.arange('x', 1, 5)
+    dg = sc.DataGroup({'a': x})
+    result = op(dg, other)
+    assert 'a' in result
+    assert sc.identical(result['a'], op(x, other))
+
+
+@pytest.mark.parametrize(
+    'op', (operator.iadd, operator.isub, operator.imul, operator.imod, operator.ipow))
+def test_inplace_data_group_with_data_group(op):
+    x = sc.arange('x', 1, 5)
+    dg1 = sc.DataGroup({'a': x.copy(), 'b': -3 * x.copy()})
+    dg2 = sc.DataGroup({'a': 2 * x.copy()})
+    op(dg1, dg2)
+
+    expected = x.copy()
+    op(expected, 2 * x)
+    assert sc.identical(dg1['a'], expected)
+    assert sc.identical(dg1['b'], -3 * x)
+
+
+# This test needs floats but imod and ipow need ints, so do this one here separately.
+def test_itruediv_data_group_with_data_group():
+    x = sc.arange('x', 1.0, 5.0)
+    dg1 = sc.DataGroup({'a': x.copy(), 'b': -3 * x.copy()})
+    dg2 = sc.DataGroup({'a': 2 * x.copy()})
+    dg1 /= dg2
+
+    assert sc.identical(dg1['a'], x / (2 * x))
+    assert sc.identical(dg1['b'], -3 * x)
+
+
+@pytest.mark.parametrize(
+    'op', (operator.iadd, operator.isub, operator.imul, operator.imod, operator.ipow))
+def test_inplace_data_group_with_data_group_extra_key(op):
+    x = sc.arange('x', 1, 5)
+    dg1 = sc.DataGroup({'a': x.copy()})
+    dg2 = sc.DataGroup({'a': 2 * x.copy(), 'b': -3 * x.copy()})
+    with pytest.raises(ValueError):
+        op(dg1, dg2)
```

Comment:
I think the most important part to check here would be that `dg1` was not modified?